src/VisualStudio/Core/Def/Implementation/LanguageClient/InProcLanguageServer.cs

```diff
@@ -79,7 +80,21 @@ public Task<InitializeResult> Initialize(JToken input, CancellationToken cancell
         }
 
         [JsonRpcMethod(Methods.InitializedName)]
-        public Task Initialized() => Task.CompletedTask;
+        public Task Initialized()
+        {
+            // Publish diagnostics for all open documents immediately following initialization.
+            var openDocuments = _workspace.GetOpenDocumentIds();
+            foreach (var documentId in openDocuments)
+            {
+                var document = _workspace.CurrentSolution.GetDocument(documentId);
+                if (document != null)
+                {
+                    PublishDiagnosticsAsync(_workspace.CurrentSolution, document);
```

Comment:
PublishDiagnosticsAsync should return a task, then can return the result of PublishDiagnosticsAsync here.  Right now it will trigger PublishDiagnosticsAsync and will return a CompletedTask before PublishDiagnosticsAsync completes.